cli/src/utils/createRustVerifyingKey.ts

```diff
@@ -140,130 +140,210 @@ async function createVerifyingKeyRsFile(
+      //   for (let j = 0; j < mydata.vk_alpha_1.length - 1; j++) {
+      //     s += "\t\t" + Array.from(mydata.vk_alpha_1[j]) + ",\n";
+      //   }
+      //   s += "\t],\n\n";
+      //   fs.writeSync(resFile, s);
+      //   s = "\tvk_beta_g2: [\n";
+      //   for (let j = 0; j < mydata.vk_beta_2.length - 1; j++) {
+      //     for (let z = 0; z < 2; z++) {
+      //       s += "\t\t" + Array.from(mydata.vk_beta_2[j][z]) + ",\n";
+      //     }
+      //   }
+      //   s += "\t],\n\n";
+      //   fs.writeSync(resFile, s);
+      //   s = "\tvk_gamme_g2: [\n";
+      //   for (let j = 0; j < mydata.vk_gamma_2.length - 1; j++) {
+      //     for (let z = 0; z < 2; z++) {
+      //       s += "\t\t" + Array.from(mydata.vk_gamma_2[j][z]) + ",\n";
+      //     }
+      //   }
+      //   s += "\t],\n\n";
+      //   fs.writeSync(resFile, s);
+
+      //   s = "\tvk_delta_g2: [\n";
+      //   for (let j = 0; j < mydata.vk_delta_2.length - 1; j++) {
+      //     for (let z = 0; z < 2; z++) {
+      //       s += "\t\t" + Array.from(mydata.vk_delta_2[j][z]) + ",\n";
+      //     }
+      //   }
+      //   s += "\t],\n\n";
+      //   fs.writeSync(resFile, s);
+      //   s = "\tvk_ic: &[\n";
+      //   let x = 0;
+
+      //   for (let ic in mydata.IC) {
+      //     s += "\t\t[\n";
+      //     for (let j = 0; j < mydata.IC[ic].length - 1; j++) {
+      //       s += "\t\t\t" + mydata.IC[ic][j] + ",\n";
+      //     }
+      //     x++;
+      //     s += "\t\t],\n";
+      //   }
+      //   s += "\t]\n};";
+      //   s += appendingString;
+
+      //   fs.writeSync(resFile, s);
+      // }
+
+      // @matteo: made it more concise and optimized (to be tested)
+      for (const path of paths) {
+        const resFile = await fs.open(path, "w");
+        let s = `use groth16_solana::groth16::Groth16Verifyingkey;
+      use anchor_lang::prelude::*;
+      
+      pub const VERIFYINGKEY: Groth16Verifyingkey = Groth16Verifyingkey {
+      \tnr_pubinputs: ${mydata.IC.length},
+      \tvk_alpha_g1: [
+      ${mydata.vk_alpha_1
+        .slice(0, -1)
+        .map((arr: any) => `\t\t${arr},`)
+        .join("\n")}
+      \t],
+      \tvk_beta_g2: [
+      ${mydata.vk_beta_2
+        .slice(0, -1)
+        .flatMap((arr: any) => arr.map((subArr: any) => `\t\t${subArr},`))
+        .join("\n")}
+      \t],
+      \tvk_gamme_g2: [
+      ${mydata.vk_gamma_2
+        .slice(0, -1)
+        .flatMap((arr: any) => arr.map((subArr: any) => `\t\t${subArr},`))
+        .join("\n")}
+      \t],
+      \tvk_delta_g2: [
+      ${mydata.vk_delta_2
+        .slice(0, -1)
+        .flatMap((arr: any) => arr.map((subArr: any) => `\t\t${subArr},`))
+        .join("\n")}
+      \t],
+      \tvk_ic: &[
+      ${mydata.IC.map(
+        (icArr: any) =>
+          `\t\t[${icArr
+            .slice(0, -1)
+            .map((item: any) => `\t\t\t${item},`)
+            .join("\n")}\n\t\t],`
+      ).join("\n")}
+      \t]
+      };
+      ${appendingString}`;
```

Comment:
Nice, I like it, much more readable now 😄

Reply:
@xonoxitron this is not covered by any tests can you write a test for this?

Reply:
sure!

Reply:
what's the status here- can it be de-commented?

Reply:
@SwenSchaeferjohann can you point me to the code line please? :)

Reply:
217 - 248 and 298 - 345 
@xonoxitron

Reply:
removed the commented code. it was an optimized version but could not deal with testing it for external reasons. 
I'll re-implement it later :)
🚀